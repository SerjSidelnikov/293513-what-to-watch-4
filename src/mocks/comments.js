export default [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,3`,
  },
  {
    id: `2`,
    author: `Amanda Greever`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,4`,
  },
  {
    id: `3`,
    author: `Matthew Lickona`,
    date: `Month dd, yyyy`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rate: `7,1`,
  },
  {
    id: `4`,
    author: `Paula Fleri-Soler`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `7,2`,
  },
  {
    id: `5`,
    author: `Paula Fleri-Soler`,
    date: `Month dd, yyyy`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rate: `7,9`,
  },
];
