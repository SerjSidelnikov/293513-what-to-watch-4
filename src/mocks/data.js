export default {
  film: {
    title: `The Grand Budapest Hotel`,
    genre: `Drama`,
    releaseDate: 2014,
  },
  movies: [
    `Fantastic Beasts`,
    `Bohemian Rhapsody`,
    `Macbeth`,
  ],
};
