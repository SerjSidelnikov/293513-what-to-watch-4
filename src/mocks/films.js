export default [
  {
    id: 1,
    title: `The Terminator`,
    poster: `https://picsum.photos/id/111/200/300`,
  },
  {
    id: 2,
    title: `Raging Bull`,
    poster: `https://picsum.photos/id/22/200/300`,
  },
  {
    id: 3,
    title: `Amelie`,
    poster: `https://picsum.photos/id/33/200/300`,
  },
  {
    id: 4,
    title: `Titanic`,
    poster: `https://picsum.photos/id/55/200/300`,
  },
  {
    id: 5,
    title: `Good Will Hunting`,
    poster: `https://picsum.photos/id/43/200/300`,
  },
  {
    id: 6,
    title: `Arrival`,
    poster: `https://picsum.photos/id/235/200/300`,
  },
  {
    id: 7,
    title: `Lost In Translation`,
    poster: `https://picsum.photos/id/122/200/300`,
  },
  {
    id: 8,
    title: `No Country For Old Men`,
    poster: `https://picsum.photos/id/109/200/300`,
  },
];
