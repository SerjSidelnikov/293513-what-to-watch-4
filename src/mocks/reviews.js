export default [
  {
    id: `0`,
    author: `Kate Muir`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `8,9`,
  },
  {
    id: `1`,
    author: `Bill Goodykoontz`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,3`,
  },
  {
    id: `2`,
    author: `Amanda Greever`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,4`,
  },
  {
    id: `3`,
    author: `Matthew Lickona`,
    date: `December 24, 2016`,
    description: `It is certainly a magical and childlike way of storytelling, even if the content is a little more adult.`,
    rating: `7,1`,
  },
  {
    id: `4`,
    author: `Paula Fleri-Soler`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `7,2`,
  },
  {
    id: `5`,
    author: `Paula Fleri-Soler`,
    date: `December 24, 2016`,
    description: `The mannered, madcap proceedings are often delightful, occasionally silly, and here and there, gruesome and/or heartbreaking.`,
    rating: `7,9`,
  },
];
