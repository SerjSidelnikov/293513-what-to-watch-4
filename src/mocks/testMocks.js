export const promoItem = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseDate: 2014,
};

export const film = {
  id: 0,
  title: `Test title`,
  poster: `https://picsum.photos/id/0`,
};

export const films = [
  {
    id: 1,
    title: `Test title 1`,
    poster: `https://picsum.photos/id/1`,
  },
  {
    id: 2,
    title: `Test title 2`,
    poster: `https://picsum.photos/id/2`,
  },
];

