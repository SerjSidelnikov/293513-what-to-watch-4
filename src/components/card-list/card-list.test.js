import React from 'react';
import renderer from 'react-test-renderer';

import CardList from './card-list';
import {films} from '../../mocks/testMocks';

describe(`CardList`, () => {
  it(`CardList rendered correctly`, () => {
    const tree = renderer.create(
        <CardList films={films}/>,
        {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
