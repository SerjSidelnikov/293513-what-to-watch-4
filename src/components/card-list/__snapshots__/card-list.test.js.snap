// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList CardList rendered correctly 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        height="100%"
        preload="none"
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      />
    </h3>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="small-movie-card__image"
    >
      <video
        autoPlay={false}
        height="100%"
        preload="none"
        width="100%"
      />
    </div>
    <h3
      className="small-movie-card__title"
    >
      <a
        className="small-movie-card__link"
        href="#"
      />
    </h3>
  </article>
</div>
`;
